### PR TITLE
Script for cleaning up vendor directory

### PR DESCRIPTION
**Pull request by ruflin**
This scripts removes the unnecessary files from the vendor directory. As we check in the vendor directory the goal is to keep the number of files to a minimum and if changes are made to the dependency make sure that it is easy to review the relevant changes.

### CODE REVIEW

File: scripts/clean_vendor.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Removes unncessary files from the vendor directories
+#
+# A aist for files to be removed is used instead of excluding files.
```
Comment (urso) [inline scripts/clean_vendor.sh]: s/aist/list/

File: scripts/clean_vendor.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# The reason is that this that this makes the setup simpler and prevents
+# from removing files by accident
+#
+# In general it should always be removed manually which files were removed.
```
Comment (urso) [inline scripts/clean_vendor.sh]: Wording?

File: scripts/clean_vendor.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# For example some projects like stretchr/testify have two LICENSE files
+# with different names, where it is ok that one is removed.
+#
+# We keep the CHANGELOG in as this makes it easy visible when updating dependencies
+# on what has changed.
+
+# In general the following files should b kept:
```
Comment (urso) [inline scripts/clean_vendor.sh]: s/b kept/be kept/

File: scripts/clean_vendor.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# * .go
+# * LICENSE* CHANGELOG* and PATENT*
```
Comment (urso) [inline scripts/clean_vendor.sh]: potentially add: Contributors, Readme ?

File: scripts/clean_vendor.sh (hunk 2 above)
Comment (ruflin) [inline scripts/clean_vendor.sh]: s/removed/checked/

File: scripts/clean_vendor.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+#
+## Removing all golang test files
+find $DIR_LIST -type f -name "*_test.go" -exec rm -r {} \;
+#
+## Removing all files starting with a dot (like .gitignore)
+find $DIR_LIST -type f -name ".*" -exec rm -r {} \;
+#
+## Removing all .txt files which are normally test data or docs
+## Excluding LICENSE as nranchev/go-libGeoIP has the license in this file
+find $DIR_LIST -type f -name "*.txt" -a ! -name "LICENSE.*" -exec rm -r {} \;
```
Comment (urso) [inline scripts/clean_vendor.sh]: let's use `-iname` in case file is named `License` or `License.txt` or ...

File: scripts/clean_vendor.sh (hunk 4 above)
Comment (urso) [inline scripts/clean_vendor.sh]: I don't see filters for CHANGELOG, PATENT in line 39.

Comment (ruflin) [inline scripts/clean_vendor.sh]: Added

**Comment by ruflin** [comment]
@urso Thanks for the review. New version pushed.

File: scripts/clean_vendor.sh (hunk 4 above)
Comment (ruflin) [inline scripts/clean_vendor.sh]: @urso That here is more a general remark. I have the LICENSES below as there was one project using it. I would only add "exception" if they occur ...

**Comment by ruflin** [comment]
@urso As discussed, all the other files we should not be deleted are now also excluded by default for *.txt files.

**Comment by ruflin** [comment]
jenkins, test it

File: scripts/clean_vendor.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# Removes unncessary files from the vendor directories
+#
+# A list for files to be removed is used instead of excluding files.
+# The reason is that this that this makes the setup simpler and prevents
```
Comment (andrewkroh) [inline scripts/clean_vendor.sh]: "that this that this"

**Pull request merged**